src/com/trollworks/gcs/character/PrerequisitesThread.java

```diff
@@ -165,31 +165,31 @@ private void processFeatures() throws Exception {
     private void buildFeatureMap(HashMap<String, ArrayList<Feature>> map, Iterator<? extends ListRow> iterator) throws Exception {
         while (iterator.hasNext()) {
             ListRow row = iterator.next();
-            if (row instanceof Equipment) {
+            if (row.isEquipment()) {
                 Equipment equipment = (Equipment) row;
                 if (!equipment.isEquipped() || equipment.getQuantity() < 1) {
                     // Don't allow unequipped equipment to affect the character
                     continue;
                 }
             }
             for (Feature feature : row.getFeatures()) {
-                processFeature(map, row instanceof Advantage ? ((Advantage) row).getLevels() : 0, feature);
-                if (feature instanceof Bonus) {
-                    ((Bonus) feature).setParent(row.toString());
+                processFeature(map, row.isAdvantage() ? ((Advantage) row).getLevels() : 0, feature);
```

Comment:
I'm not sure that I'm a fan of this change you've sprinkled throughout, i.e. additional of an `isXXX()` method rather than using `instanceof`. This code right here illustrates why: it makes an assumption that, if `isAdvantage()` returns true, that the object can be cast to an `Advantage`, which **may not** be true using this approach, as any arbitrary row can now have that return true in the future.

That said, I do agree that it is slightly easier to type `row.isAdvantage()` vs `row instanceof Advantage`... but not so much so that I'd make this particular change.

Reply:
You are correct.   I was on a coding fling and was making changes Willy nilly.    Since they are not required for the feature to work, I will remove them.